platforms/hyperledger-indy/configuration/roles/setup/identity/tasks/nested_main.yaml

```diff
@@ -0,0 +1,119 @@
+- name: Select Admin Identity for Organisation {{ component_name }}
+  shell: |
+  
+    selectedAdmin=""
+    first_global_trustee=""
+    admin_Org=""
+    global_Org=""
+    admin_url=""
+    global_url=""
+    {% if network['organizations'] is defined %}
+      {% for organization in network['organizations'] %}
+        first_trustee_in_org=""
+        {% if organization.services.trustees is defined %}
+          {% for trustee in organization.services.trustees %}
+            if [ -z "$first_trustee_in_org" ]
+            then
+              if [ organization.name == "{{ component_name }}" ]
+              then
+                first_trustee_in_org="{{ trustee.name }}"
+                admin_Org="{{ organization.name }}"
+                admin_url="{{ organization.vault.url }}"
+              fi
+            fi
+            if [ -z "$first_global_trustee" ]
+            then
+              first_global_trustee="{{ trustee.name }}"
+              global_Org="{{ organization.name }}"
+              global_url="{{ organization.vault.url }}"
+            fi
+          {% endfor %}
+        {% endif %}
+        
+        {% if organization.services.stewards is defined %}
+          {% for steward in organization.services.stewards %}
+            if [ -z "$first_trustee_in_org" ]
+            then
+              if [ organization.name == "{{ component_name }}" ]
+              then
+                first_trustee_in_org="{{ steward.name }}"
+                admin_Org="{{ organization.name }}"
+                admin_url="{{ organization.vault.url }}"
+              fi
+            fi
+            if [ -z "$first_global_trustee" ]
+            then
+              first_global_trustee="{{ steward.name }}"
+              global_Org="{{ organization.name }}"
+              global_url="{{ organization.vault.url }}"
+            fi
+          {% endfor %}
+        {% endif %}
+      {% endfor %}
+    {% endif %}
+    
+    if [ -z "$first_trustee_in_org" ]
+      then
+        selectedAdmin="${first_trustee_in_org}"
+        adminUrl="${admin_url}"
+        adminOrg="${admin_Org}"
+      else
+        selectedAdmin="${first_global_trustee}"
+        adminUrl="${global_url}"
+        adminOrg="${global_Org}"
+    fi
+    echo "${selectedAdmin}" >> admin.yaml
+    echo "${adminUrl}" >> admin.yaml
+    echo "${adminOrg}" >> admin.yaml
+    
+  register: admin.yaml
+
+- name: Debugging
+  debug:
+    msg: "the value file is {{lookup('file', 'admin.yaml') }}"
+    
+- name: "Inserting into file"
+  include_vars:
+    file: admin.yaml
+    name: admin_var
+    
+# Create Deployment files for new Identities
+- name: "Calling Helm Release Development Role..."
+  include_role:
+    name: ledger_txn
```

Comment:
Here https://github.com/hyperledger-labs/blockchain-automation-framework/blob/0cc89d24d0f27070ee9bc256b2693018bc645100/platforms/hyperledger-indy/configuration/roles/create/helm_component/ledger_txn/tasks/nested_main.yaml#L8
you assume that the admin identity is always a trustee - it can be also a steward, with a different vault path